arviz/tests/test_stats.py

```diff
@@ -334,6 +349,133 @@ def test_multidimensional_log_likelihood(func):
     assert_array_almost_equal(frm[:4], fr1[:4])
 
 
+@pytest.mark.parametrize(
+    "args",
+    [
+        {"y": "obs"},
+        {"y": "obs", "y_hat": "obs"},
+        {"y": "arr", "y_hat": "obs"},
+        {"y": "obs", "y_hat": "arr"},
+        {"y": "arr", "y_hat": "arr"},
+        {"y": "obs", "y_hat": "obs", "log_weights": "arr"},
+        {"y": "arr", "y_hat": "obs", "log_weights": "arr"},
+        {"y": "obs", "y_hat": "arr", "log_weights": "arr"},
+        {"idata": False},
+    ],
+)
+def test_loo_pit(centered_eight, args):
+    y = args.get("y", None)
+    y_hat = args.get("y_hat", None)
+    log_weights = args.get("log_weights", None)
+    y_arr = centered_eight.observed_data.obs
+    y_hat_arr = centered_eight.posterior_predictive.obs.stack(samples=("chain", "draw"))
+    log_like = centered_eight.sample_stats.log_likelihood.stack(samples=("chain", "draw"))
+    n_samples = len(log_like.samples)
+    ess_p = ess(centered_eight.posterior, method="mean")
+    reff = np.hstack([ess_p[v].values.flatten() for v in ess_p.data_vars]).mean() / n_samples
+    log_weights_arr = psislw(-log_like, reff=reff)[0]
+
+    if args.get("idata", True):
+        if y == "arr":
+            y = y_arr
+        if y_hat == "arr":
+            y_hat = y_hat_arr
+        if log_weights == "arr":
+            log_weights = log_weights_arr
+        loo_pit_data = loo_pit(idata=centered_eight, y=y, y_hat=y_hat, log_weights=log_weights)
+    else:
+        loo_pit_data = loo_pit(idata=None, y=y_arr, y_hat=y_hat_arr, log_weights=log_weights_arr)
+    assert np.all((loo_pit_data >= 0) & (loo_pit_data <= 1))
+
+
+def test_loo_pit_bad_input(centered_eight):
+    arr = np.random.random((8, 200))
+    with pytest.raises(ValueError):
+        loo_pit(idata=arr, y="obs")
+    with pytest.raises(ValueError):
+        loo_pit(idata=None, y="obs")
+    with pytest.raises(ValueError):
+        loo_pit(idata=centered_eight, y=arr, y_hat=None)
+
+
+@pytest.mark.parametrize("arg", ["y", "y_hat", "log_weights"])
+def test_loo_pit_bad_input_type(centered_eight, arg):
+    kwargs = {"y": "obs", "y_hat": "obs", "log_weights": None}
+    kwargs[arg] = 2
+    with pytest.raises(ValueError):
+        loo_pit(idata=centered_eight, **kwargs)
+
+
+def test_loo_pit_bad_input_shape():
+    y = np.random.random(8)
+    y_hat = np.random.random((8, 200))
+    log_weights = np.random.random((8, 200))
+    with pytest.raises(ValueError):
+        loo_pit(y=y, y_hat=y_hat[None, :], log_weights=log_weights)
+    with pytest.raises(ValueError):
+        loo_pit(y=y, y_hat=y_hat[1:3, :], log_weights=log_weights)
+    with pytest.raises(ValueError):
+        loo_pit(y=y, y_hat=y_hat[:, :100], log_weights=log_weights)
+
+
+@pytest.mark.parametrize("group", ["both", "observed_data", "posterior_predictive"])
```

Comment:
The combinatorics on the parameterization here mean that a whole lot of tests are being run. Do we need to test for every combination of variation? If not would it be easier to enumerate the minimum number of tests needed by hand?

It might not be just asking

Reply:
Probably not all cases are extremely necessary, but whatever the subset, some errors may not trigger a test failure. It is also not the most important function though, so they can probably be reduced.